src/main/java/org/joda/pa/MetaBean.java

```diff
@@ -103,7 +105,7 @@ static MetaBean of(Class<?> cls) {
      * 
      * @return the stream of properties on the bean, not null
      */
-    Stream<MetaProperty<?>> metaProperties();
+    Stream<MetaProperty<B, ?>> metaProperties();
```

Comment:
This is where it starts to get complicated - object hierarchies. An `Address` might define certain meta-properties, and a subclass `CompanyAddress` might define some more meta-properties. What is the B type of the meta-properties of `CompanyAddress`? Saying that B must always be `CompanyAddress` implies creating a meta-property instance for all properties of `CompanyAddress` and `Address`. Whereas it would be useful to be able to inherit the parent meta-properties but use them against the child. IIRC, generics are not well setup to deal with this, but it would be interesting to try. This will affect signatures on `MetaProperty` as well as here.